library/src/include/tree_node.h

```diff
@@ -92,8 +93,32 @@ class TreeNode
             batch     = p->batch;
             direction = p->direction;
         }
+
+        Pow2Lengths1Single.insert(std::make_pair(8192, 64));
+        Pow2Lengths1Single.insert(std::make_pair(16384, 64));
+        Pow2Lengths1Single.insert(std::make_pair(32768, 128));
+        Pow2Lengths1Single.insert(std::make_pair(65536, 256));
+        Pow2Lengths1Single.insert(std::make_pair(131072, 64));
+        Pow2Lengths1Single.insert(std::make_pair(262144, 64));
+
+        Pow2Lengths1Double.insert(std::make_pair(4096, 64));
+        Pow2Lengths1Double.insert(std::make_pair(8192, 64));
+        Pow2Lengths1Double.insert(std::make_pair(16384, 64));
+        Pow2Lengths1Double.insert(std::make_pair(32768, 128));
+        Pow2Lengths1Double.insert(std::make_pair(65536, 64));
+        Pow2Lengths1Double.insert(std::make_pair(131072, 64));
     }
 
+    // Maps from length[0] to divLength1 for 1D transforms in
+    // single and double precision for power-of-two transfor sizes
+    // using blocks.
+    std::map<size_t, size_t> Pow2Lengths1Single;
```

Comment:
Good for now.  I still think subclass for each instance of float and double would be better.